esp-radio/src/compat/malloc.rs

```diff
@@ -35,13 +35,54 @@ mod esp_alloc {
 
     unsafe impl Allocator for InternalMemory {
         fn allocate(&self, layout: Layout) -> Result<NonNull<[u8]>, AllocError> {
-            let raw_ptr = unsafe { super::malloc_internal(layout.size()) };
-            let ptr = NonNull::new(raw_ptr).ok_or(AllocError)?;
+            // We assume malloc returns a 4-byte aligned pointer. We can skip aligning types
```

Comment:
This alignment code isn't necessary for esp-radio, but I've copied it from esp-rtos to keep them the same.